src/main/scala/rules/SymbolicExecutionRules.scala

```diff
@@ -45,9 +45,19 @@ trait SymbolicExecutionRules {
           case Some(trafo) => trafo.f(ce)
           case _ => ce
         }
-        res.counterexample = Some(finalCE)
-      }
-    }
+        Some(finalCE)
+      } else None
+    } else None
+
+    val branchconditions = if (Verifier.config.enableBranchconditionReporting()) {
+      v.decider.pcs.branchConditionExps.flatten
+        .filterNot(e => e.isInstanceOf[viper.silver.ast.TrueLit]) /* remove "true" bcs introduced by viper.silicon.utils.ast.BigAnd */
```

Comment:
Could you alternatively change `BigAnd` to avoid such `true`s in the first place?